src/lib/utilities/route-for-api.ts

```diff
@@ -99,34 +103,34 @@ export function pathForApi(
     settings: '/settings',
     'task-queue': `/namespaces/${parameters?.namespace}/task-queues/${parameters?.queue}`,
-    'task-queue.compatibility': `/namespaces/${parameters?.namespace}/task-queues/${parameters?.queue}/compatibility`,
+    'task-queue.compatibility': `/namespaces/${parameters?.namespace}/task-queues/${parameters?.queue}/worker-build-id-compatibility`,
     user: '/me',
     'worker-task-reachability': `/namespaces/${parameters?.namespace}/worker-task-reachability`,
-    'workflow.terminate': `/namespaces/${parameters?.namespace}/workflows/${parameters?.workflowId}/runs/${parameters?.runId}/terminate`,
-    'workflow.cancel': `/namespaces/${parameters.namespace}/workflows/${parameters.workflowId}/runs/${parameters.runId}/cancel`,
-    'workflow.signal': `/namespaces/${parameters.namespace}/workflows/${parameters.workflowId}/runs/${parameters.runId}/signal`,
-    'workflow.reset': `/namespaces/${parameters.namespace}/workflows/${parameters.workflowId}/runs/${parameters.runId}/reset`,
-    workflow: `/namespaces/${parameters?.namespace}/workflows/${parameters?.workflowId}/runs/${parameters?.runId}`,
-    'workflows.archived': `/namespaces/${parameters?.namespace}/workflows/archived`,
+    'workflow.terminate': `/namespaces/${parameters?.namespace}/workflows/${parameters?.workflowId}/terminate`,
+    'workflow.cancel': `/namespaces/${parameters.namespace}/workflows/${parameters.workflowId}/cancel`,
+    'workflow.signal': `/namespaces/${parameters.namespace}/workflows/${parameters.workflowId}/signal/${parameters.signalName}`,
+    'workflow.reset': `/namespaces/${parameters.namespace}/workflows/${parameters.workflowId}/reset`,
+    workflow: `/namespaces/${parameters?.namespace}/workflows/${parameters?.workflowId}`,
+    'workflows.archived': `/namespaces/${parameters?.namespace}/archived-workflows`,
     workflows: `/namespaces/${parameters?.namespace}/workflows`,
-    'workflows.count': `/namespaces/${parameters?.namespace}/workflows/count`,
-    'activity.complete': `/namespaces/${parameters.namespace}/workflows/${parameters.workflowId}/runs/${parameters.runId}/activities/${parameters.activityId}/complete`,
-    'activity.fail': `/namespaces/${parameters.namespace}/workflows/${parameters.workflowId}/runs/${parameters.runId}/activities/${parameters.activityId}/fail`,
-    'batch-operations': `/namespaces/${parameters.namespace}/batch-operations`,
-    'batch-operation.describe': `/namespaces/${parameters.namespace}/batch-operations/describe`,
+    'workflows.count': `/namespaces/${parameters?.namespace}/workflow-count`,
+    'activity.complete': `/namespaces/${parameters.namespace}/activities/complete-by-id`,
+    'activity.fail': `/namespaces/${parameters.namespace}/activities/fail-by-id`,
+    'batch-operations': `/namespaces/${parameters.namespace}/batch-operations/${parameters?.batchJobId}`,
+    'batch-operation.describe': `/namespaces/${parameters.namespace}/batch-operations/${parameters?.batchJobId}`,
```

Comment:
Since these are now identical, maybe we just keep one? Also just so I understand, to create Batch Operation, a POST is sent to `/namespaces/<namespace>/batch-operations/<jobId>` ? Does the jobId also need to be included in the body of the request?

Reply:
Good catch. And I don't believe so: it should be extracted by our GRPC gateway